packages/shared/lib/services/connection.service.ts

```diff
@@ -483,12 +481,11 @@ class ConnectionService {
             .update({ connection_config: config });
     }
 
-    public async updateMetadata(connection: Connection, metadata: Metadata): Promise<Metadata> {
-        const existingMetadata = await this.getMetadata(connection);
-        const newMetadata = { ...existingMetadata, ...metadata };
-        await this.replaceMetadata(connection, newMetadata);
-
-        return newMetadata;
+    public async updateMetadata(connections: Connection[], metadata: Metadata): Promise<void> {
+        for (const connection of connections) {
+            const newMetadata = { ...connection.metadata, ...metadata };
+            await this.replaceMetadata([connection.id as number], newMetadata);
```

Comment:
would have been nice to have all the queries as part of a transaction. Here we might potentially have some out of sync Metadata if one or more query fails somehow

Reply:
Updated with https://github.com/NangoHQ/nango/pull/2145/commits/25669dc4c4d26e89079a106842d37dfd8ac514b6